### PR TITLE
added annotated StringJoiner.java and a test file to support the chan…

### PR DESCRIPTION
**Pull request by kanha95**
fixes Issue #1147

### CODE REVIEW

File: checker/tests/nullness/Issue1147.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import java.util.StringJoiner;
+
+class Issue1147 {
+
+    public static void main(String[] args) {
+
+        StringJoiner sj = new StringJoiner(",");
+
+        sj.add("a");
+        // Nullness Checker raises false positive warning
```
Comment (mernst) [inline checker/tests/nullness/Issue1147.java]: Please remove this comment because it won't be true once your pull request is merged.

Comment (kanha95) [inline checker/tests/nullness/Issue1147.java]: sure

**Comment by kanha95** [comment]
@mernst Please review the codes and suggest why it is failing in case of jdk8 test.

**Comment by mernst** [comment]
Since `@NonNull` is the default, those annotations are not necessary.  If you aren't going to include the whole original file, there is no need for private members.  It seems that the `add` method is the only one needed.

**Comment by kanha95** [comment]
Yes, Right. I will fix it up. Is that the reason for failure with jdk 8 test? i don't think so.

**Comment by mernst** [comment]
Currently, to same time during builds, Travis-CI does not rebuild the JDK but downloads a previously-built one.
This makes builds much faster, but it suffers spurious failures for any build that adds a test for JDK changes. :-(

**Comment by kanha95** [comment]
@mernst removing the @NonNull tags generates error. I think @NonNull is not set as default type here as StringJoiner class doesn't take any Type parameters. So by default it's extending the  @Nullable Object. The previously annotated code was fine though. Should i revert the changes back?

**Comment by mernst** [comment]
Could you be more specific?  What new error results from removing the `@NonNull` annotations, that does not result from having them present?

**Comment by kanha95** [comment]
@mernst 
I am not sure professor. I am currently analyzing the results thrown by travis build. They make no sense at all. Errors & warnings are forecasted out of some unusual method bodies which are not present in my source code.

**Comment by mernst** [comment]
When you created this pull request, did you allow changes to your fork, as described here:
https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/
If not, could you do so?  Thanks.

**Comment by mernst** [comment]
Did you ensure that all the tests work locally before pushing them to your branch?

**Comment by mernst** [comment]
Can you pull upstream into your branch or give the Checker Framework maintainers permission to do so?

**Comment by kanha95** [comment]
@mernst  Sorry for late reply. I was out for my college. Yes i had checked them locally professor. I have allowed edits from the maintainers.

**Comment by mernst** [comment]
I pulled upstream into your branch.  I am hopeful that this will fix the tests; we'll see.

**Comment by kanha95** [comment]
Thanks again !!! I am sorry for the branch name. i would definitely write descriptive branch names in my future pull requests.

**Pull request merged**